CondCore/BeamSpotPlugins/plugins/BeamSpotOnline_PayloadInspector.cc

```diff
@@ -78,28 +84,85 @@ namespace {
+          case endTime:
+            return this->m_payload->GetEndTime();
+          case lumiRange:
+            return this->m_payload->GetLumiRange();
           default:
             return ret;
         }
       };
 
-      unsigned int yBin = 6;
-      for (int foo = parameters::lastLumi; foo != parameters::END_OF_TYPES; foo++) {
+      //_____________________________________________________________________________
+      std::function<cond::Time_t(parameters)> myTimeFunctor = [this](parameters my_param) {
+        cond::Time_t ret(1);
+        switch (my_param) {
+          case creationTime:
+            return this->m_payload->GetCreationTime();
+          case startTimeStamp:
+            return this->m_payload->GetStartTimeStamp();
+          case endTimeStamp:
+            return this->m_payload->GetEndTimeStamp();
+          default:
+            return ret;
+        }
+      };
+
+      unsigned int yBin = span;
+      for (int foo = parameters::lastLumi; foo != parameters::startTime; foo++) {
         parameters param = static_cast<parameters>(foo);
         std::string theLabel = this->getStringFromTypeEnum(param);
         h2_ExtraBSParameters->GetYaxis()->SetBinLabel(yBin, theLabel.c_str());
 
-        edm::LogInfo("BeamSpotOnline_PayloadInspector")
-            << theLabel.c_str() << " : " << mycutFunctor(param) << std::endl;
-
-        if (param == BeamSpotPI::creationTime) {
-          h2_ExtraBSParameters->SetBinContent(1, yBin, m_payload->GetCreationTime());
-        } else {
-          h2_ExtraBSParameters->SetBinContent(1, yBin, mycutFunctor(param));
+        if (foo <= parameters::maxPVs) {
+          const auto output = try_<int, std::out_of_range>(std::bind(myIntFunctor, param), print_error);
+          edm::LogInfo("BeamSpotOnline_PayloadInspector") << theLabel.c_str() << " : " << output << std::endl;
+          h2_ExtraBSParameters->SetBinContent(1, yBin, output);
+        } else if (foo <= parameters::rmsErrorPV) {
+          const auto output = try_<float, std::out_of_range>(std::bind(myFloatFunctor, param), print_error);
+          edm::LogInfo("BeamSpotOnline_PayloadInspector") << theLabel.c_str() << " : " << output << std::endl;
+          h2_ExtraBSParameters->SetBinContent(1, yBin, output);
+        } else if (foo <= parameters::endTimeStamp) {
+          const auto output = try_<cond::Time_t, std::out_of_range>(std::bind(myTimeFunctor, param), print_error);
+          edm::LogInfo("BeamSpotOnline_PayloadInspector") << theLabel.c_str() << " : " << output << std::endl;
+          h2_ExtraBSParameters->SetBinContent(1, yBin, output);
+          //} else if( foo <=parameters::lumiRange){
+          // const auto output = try_<std::string,std::out_of_range>(std::bind(myStringFunctor, param), print_error);
+          //edm::LogInfo("BeamSpotOnline_PayloadInspector") << theLabel.c_str() << " : " << output << std::endl;
+          //h2_ExtraBSParameters->SetBinContent(1, yBin, output);
```

Comment:
I assume the fact that you didnt remove this means that it could be useful for something in the future?

Reply:
Yes, it would actually need some way to display strings in a TH2F , perhaps using some adjusted `TLatex` box within an otherwise empty bin. I am open for suggestions.

Reply:
I don't have any good idea about this at the moment.
I'll sign the PR for now so that we have the PI working properly with most of the new members of BeamSpotOnline and maybe in the future we understand how to nicely display strings in a TH2F. 😄